docs/apis/streaming/windows.md

```diff
@@ -24,1023 +24,593 @@ specific language governing permissions and limitations
-        <td>
-          <p>
-          Defines a window of 1000 elements, that "tumbles". This means that elements are
-          grouped according to their arrival time (equivalent to processing time) in groups of 1000 elements,
-          and every element belongs to exactly one window.
-    {% highlight scala %}
-keyedStream.countWindow(1000)
-    {% endhighlight %}
-        </p>
-        </td>
-      </tr>
-      <tr>
-      <td><strong>Sliding count window</strong><br>KeyedStream &rarr; WindowedStream</td>
-      <td>
-        <p>
-          Defines a window of 1000 elements, that "slides" every 100 elements. This means that elements are
-          grouped according to their arrival time (equivalent to processing time) in groups of 1000 elements,
-          and every element can belong to more than one window (as windows overlap by at most 900 elements).
-  {% highlight scala %}
-keyedStream.countWindow(1000, 100)
-  {% endhighlight %}
-        </p>
-      </td>
-    </tr>
-  </tbody>
-</table>
-
+input
+    .keyBy(<key selector>)
+    .window(<window assigner>)
+    .<windowed transformation>(<window function>)
+{% endhighlight %}
 </div>
 </div>
 
-### Advanced Window Constructs
+We will cover the different window assigners in [window assigners](#window-assigners).
+
+The window transformation can be one of `reduce()`, `fold()` or `apply()`. Which respectively
+takes a `ReduceFunction`, `FoldFunction` or `WindowFunction`. We describe each of these ways
+of specifying a windowed transformation in detail in [window functions](#window-functions).
+
+For more advanced use cases you can also specify a `Trigger` that determines when exactly a window
+is being considered as *ready for processing*. These will be covered in more detail in
+[triggers](#triggers).
+
+## Window Assigners
+
+The window assigner specifies how elements of the stream are divided into slices. You can provide
+your custom window assigner by implementing `WindowAssigner` but Flink comes with
+window assigners for typical use cases: *tumbling windows*, *sliding windows*, *session windows*
+and *global windows*. Except for the last, all of these assign windows based on time, which
+can either be processing time or event time.
+
+Let's first look at how each of these window assigners works before looking at how they can be used
+in a Flink program. We will be using abstract figures to visualize the workings of each assigner:
+in the following, the purple circles are elements of the stream, they are partitioned
+by some key (in this case *user 1*, *user 2* and *user 3*) and the x-axis shows the progress
+of time.
+
+### Global Windows
+
+Global windows are a way of specifying that we don't want to subdivide our elements into windows.
+Each element is assigned to the one single *global window* (still separate for each key, of course).
+This is only useful if you also specify a custom [trigger](#triggers), otherwise you will never
```

Comment:
This windowing scheme is only useful if you also specify a custom [trigger](#triggers). Otherwise, no computation is ever going to be performed, as the global window does not have a natural end at which we could process
the aggregated elements.